test/Stream/MemoryStreamReader.test.cpp

```diff
@@ -95,3 +97,36 @@ TEST_F(SimpleMemoryReader, SeekRelativeOutOfBoundsEndPreservesPosition) {
 	EXPECT_THROW(stream.SeekForward(6), std::runtime_error);
 	EXPECT_EQ(position, stream.Position());
 }
+
+TEST(MemoryReader, ReadNullTerminatedString)
+{
+	std::array<char, 5> buffer{ 'n', 'u', 'l', 'l', '\0' };
+	std::array<char, 5> result;
+
+	// Check null-terminated
+	Stream::MemoryReader reader(&buffer[0], 5);
+	std::string str;
+	
+	EXPECT_NO_THROW(str = reader.ReadNullTerminatedString());
+
+	// Copy to array
+	std::copy(str.begin(), str.end(), result.data());
+	
+	// Ensure null-terminator was not copied
+	EXPECT_NE(buffer, result);
```

Comment:
This check makes an assumption about the last character of the `result` buffer being non-zero, however, the last character is actually left uninitialized, and so could be zero. That's likely the cause of the test failures.

Testing using the `std::string` result directly, rather than copying to a `std::array` may sidestep this problem.